lib/APPTurtle/src/App.cpp

```diff
@@ -295,57 +294,38 @@ void App::handleTurtle()
     /* Check for new data. */
     if (true == m_isNewTurtleSpeedSetpoint)
     {
-        /*
-        Max speed is currently unknown to DCS, but using a big value will ensure that
-        RU drives as fast as possible as it will cap the speed to its own maximum.
-        */
-        const int16_t MOTOR_MAX_SPEED = 5000;
-        SpeedData     payload;
-
-        /*
-        Move turtle regardless of the exact speed setpoint received.
-        */
-        if (m_turtleSpeedSetpoint.linear.x > 0.0F)
-        {
-            payload.left  = MOTOR_MAX_SPEED;
-            payload.right = MOTOR_MAX_SPEED;
-        }
-        /* Turn left. */
-        else if (m_turtleSpeedSetpoint.angular.z > 0.0F)
-        {
-            payload.left  = -MOTOR_MAX_SPEED;
-            payload.right = MOTOR_MAX_SPEED;
-        }
-        /* Turn right. */
-        else if (m_turtleSpeedSetpoint.angular.z < 0.0F)
-        {
-            payload.left  = MOTOR_MAX_SPEED;
-            payload.right = -MOTOR_MAX_SPEED;
-        }
+        TurtleSpeed payload;
+        int32_t     linearSpeed  = m_turtleSpeedSetpoint.linear.x * 1000;  /* Linear speed in mm/s */
+        int32_t     angularSpeed = m_turtleSpeedSetpoint.angular.z * 1000; /* Angular speed in mrad/s */
+
+        payload.linearCenter = linearSpeed;
+        payload.angular      = angularSpeed;
+
+        LOG_DEBUG("Linear speed: %d mm/s, Angular speed: %d mrad/s", payload.linearCenter, payload.angular);
 
-        if (false == m_smpServer.sendData(m_serialMuxProtChannelIdMotorSpeeds, &payload, sizeof(payload)))
+        if (false == m_smpServer.sendData(m_serialMuxProtChannelIdTurtle, &payload, sizeof(payload)))
         {
             LOG_WARNING("Failed to send motor speeds to RU.");
```

Comment:
Again: Error Handling in this case ? Stop RU via reset, RED led flash ?

Reply:
I do not believe this is a fatal error and the system should be stopped. RU already stops if communication is lost, so it is not a safety concern. And stopping the system would prevent any chance at retrying at a later time

Reply:
ok